### PR TITLE
doc(sm): document self-manage troubleshooting scripts

### PR DESCRIPTION
**Pull request by leiicamundi**
## Description

This PR introduces a new section in the Troubleshooting documentation titled "Anomaly Detection Scripts". It provides an overview of the newly added scripts in the [c8-sm-checks](https://github.com/camunda/c8-sm-checks) project, which aim to help in the detection of anomalies within C8 deployments for self-managed installations. 

## When should this change go live?

- [x] This change is not yet live and should not be merged until the [c8-sm-checks](https://github.com/camunda/c8-sm-checks)  repository is public and stable - this is tracked by https://github.com/camunda/c8-sm-checks/issues/4
- [x] There is no urgency with this change.

## PR Checklist

- [x] I have added changes to the relevant `/versioned_docs` directory, or they are not for an **already released version**.
- [x] I have added changes to the main `/docs` directory (aka `/next/`), or they are not for **future versions**.
- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned an engineering manager or tech lead as a reviewer, or my changes do not require an Engineering review.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @christinaausley as a reviewer, or my changes do not require a technical writer review.

### CODE REVIEW

**Review by Langleu — DISMISSED** [review]
looks good content wise from my side 👍. 
Good job 🚀

File: versioned_docs/version-8.5/self-managed/operational-guides/troubleshooting/troubleshooting.md [hunk 1]
```diff
@@ -66,3 +66,81 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
...
+
+For example:
+
+```
+[KO] None of the ingresses contain the annotation nginx.ingress.kubernetes.io/backend-protocol: GRPC, which is required for Zeebe ingress.
+```
+
+The error message suggests adjusting the ingress configuration to include the required annotation. One can also explore the source of the script to have a better understanding of the reason for the failure.
+
+Please also note that sometimes, some checks may not be applicable to your setup if it's custom (e.g., the ingress may be different).
```
Comment (Langleu) [inline versioned_docs/version-8.5/self-managed/operational-guides/troubleshooting/troubleshooting.md:146]: `e.g.` replace with `for example`, see [related message](https://camunda.slack.com/archives/C026U8GBNSW/p1709922869101609)

File: docs/self-managed/operational-guides/troubleshooting/troubleshooting.md [hunk 2]
```diff
@@ -66,3 +66,81 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
...
+
+For example:
+
+```
+[KO] None of the ingresses contain the annotation nginx.ingress.kubernetes.io/backend-protocol: GRPC, which is required for Zeebe ingress.
+```
+
+The error message suggests adjusting the ingress configuration to include the required annotation. One can also explore the source of the script to have a better understanding of the reason for the failure.
+
+Please also note that sometimes, some checks may not be applicable to your setup if it's custom (e.g., the ingress may be different).
```
Comment (Langleu) [inline docs/self-managed/operational-guides/troubleshooting/troubleshooting.md:146]: `e.g.` replace with `for example`, see [related message](https://camunda.slack.com/archives/C026U8GBNSW/p1709922869101609)

**Review by Langleu — APPROVED** [review]

**Review by Langleu — DISMISSED** [review]

File: docs/self-managed/operational-guides/troubleshooting/troubleshooting.md [hunk 3]
```diff
@@ -66,3 +66,87 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
...
+
+```bash
+./checks/zeebe/connectivity.sh -a https://local.distro.example.com/auth/realms/camunda-platform/protocol/openid-connect/token -i myclientid -s 0Rn28VrQxGNxowrCWe6wbujwFghO4990 -u zeebe.distro.example.com -H zeebe.local.distro.example.com:443
+```
+
+Find more information on [how to register your application on Identity](https://github.com/camunda-community-hub/camunda-8-examples/blob/main/payment-example-process-application/kube/README.md#4-generating-an-m2m-token-for-our-application).
+
+### Interpretation of the results
+
+Each script produces an output indicating the status of individual checks, which can be either `[OK]`, which signals a healthy status, or `[KO]`, which signals an unhealthy status.
```
Comment (christinaausley) [inline docs/self-managed/operational-guides/troubleshooting/troubleshooting.md:128]: I added a bit to this sentence, WDYT?

**Review by christinaausley — DISMISSED** [review]

**Review by christinaausley — APPROVED** [review]
I like this change -- helps differentiate for sure.

File: docs/self-managed/operational-guides/troubleshooting/troubleshooting.md [hunk 4]
```diff
@@ -66,3 +66,87 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
...
+
+The [c8-sm-checks](https://github.com/camunda/c8-sm-checks) project introduces a set of scripts to aid detection of Camunda deployment anomalies.
+
+These scripts perform health checks on various aspects of the Kubernetes installation and Zeebe components, providing insights into potential issues that may affect the performance or stability.
+
+### Usage
+
+Each script in the `c8-sm-checks` project can be executed independently, allowing you to target specific areas for troubleshooting and verification.
+
+To utilize these scripts effectively, ensure you have the necessary permissions and access to your Kubernetes cluster. Additionally, make sure you have the required dependencies installed on your system, such as `kubectl`, `helm`, `curl`, and `grpcurl`.
```
Comment (Langleu) [inline docs/self-managed/operational-guides/troubleshooting/troubleshooting.md:80]: I'd probably rephrase this part about `necessary permissions` since we don't mention anything what the minimal set of permissions is so just saying make sure to have something of which no one knows what it is, is not gonna help.

Probably rephrase it to just the following or something similar.

```
.., ensure you have access to your Kubernetes cluster.
```

File: versioned_docs/version-8.5/self-managed/operational-guides/troubleshooting/troubleshooting.md [hunk 5]
```diff
@@ -66,3 +66,87 @@ A gateway timeout can occur if the headers of a response are too big (for exampl
...
+
+The [c8-sm-checks](https://github.com/camunda/c8-sm-checks) project introduces a set of scripts to aid detection of Camunda deployment anomalies.
+
+These scripts perform health checks on various aspects of the Kubernetes installation and Zeebe components, providing insights into potential issues that may affect the performance or stability.
+
+### Usage
+
+Each script in the `c8-sm-checks` project can be executed independently, allowing you to target specific areas for troubleshooting and verification.
+
+To utilize these scripts effectively, ensure you have the necessary permissions and access to your Kubernetes cluster. Additionally, make sure you have the required dependencies installed on your system, such as `kubectl`, `helm`, `curl`, and `grpcurl`.
```
Comment (Langleu) [inline versioned_docs/version-8.5/self-managed/operational-guides/troubleshooting/troubleshooting.md:80]: same as above about the permissions part.

**Review by Langleu — CHANGES_REQUESTED** [review]
some small minor rephrasing requests.

File: docs/self-managed/operational-guides/troubleshooting/troubleshooting.md (hunk 4 above)
Comment (Langleu) [reply docs/self-managed/operational-guides/troubleshooting/troubleshooting.md:80]: makes sense 👍 , thanks for the adjustments!

**Review by Langleu — DISMISSED** [review]

**Review by christinaausley — APPROVED** [review]
